### ISSUE TITLE
executable can't find it's dependencies when installed with npm

### ISSUE THREAD

**Issue by joprice** [alexa/ask-cli#42]
## I'm submitting a...

<pre><code>
[ ] Regression (a behavior that used to work and stopped working in a new release)
[x] Bug report  
[ ] Performance issue
[ ] Feature request
[ ] Documentation issue or request
[ ] Other... Please describe:
</code></pre>

I get the error `Error: Cannot find module '@src/utils/constants'` when trying to run the cli after install through npm.

## Expected Behavior

It can find its own dependencies

## Current Behavior

It can't find its own dependencies:

```
./node_modules/.bin/ask
module.js:550
    throw err;
    ^

Error: Cannot find module '@src/utils/constants'
    at Function.Module._resolveFilename (module.js:548:15)
    at Function.Module._resolveFilename (/Users/josephprice/dev/github.com/iheartradio/ce-alexa-santa-skill/lambda/node_modules/module-alias/index.js:49:29)
    at Function.Module._load (module.js:475:25)
    at Module.require (module.js:597:17)
    at require (internal/module.js:11:18)
    at Object.<anonymous> (/Users/josephprice/dev/github.com/iheartradio/ce-alexa-santa-skill/lambda/node_modules/ask-cli/lib/commands/init/index.js:1:81)
    at Module._compile (module.js:653:30)
    at Object.Module._extensions..js (module.js:664:10)
    at Module.load (module.js:566:32)
    at tryModuleLoad (module.js:506:12)
```

## Steps to Reproduce (for bugs)

* Install ask-cli as a dev dependency. In my package.json, it resolved to "^1.7.18".
* Try to use it by executing `./node_modules/.bin/ask`

## Your Environment and Context

* ask-cli version: 1.7.18
* Operating System and version: mac os 10.12.6
* Node.js version used for development: v10.10.0
* NPM version used for development: 6.13.4

**Comment by habuma** [comment]
I'm guessing that you installed ASK CLI like this:

```
$ npm install ask-cli
```

That will install it into the current node project as if it were a library. But that's not how you should install it. Instead, you should install it with `-g` so that it is available as a command. (Also on most Unix OS's you'll likely need to `sudo`). Like this:

```
$ sudo npm install -g ask-cli
```

Then you should be able to run the ASK CLI without a path like this:

```
$ ask -v
```

**Comment by joprice** [comment]
I installed it with `--save-dev`. I would like to manage the version as a dev dependency, so that I can pin the version used by a given project for stability. I don't see why this project can't be installed and run like many other tools under node_modules, making hermetic builds and vendoring possible.

**Comment by Shreyas-vgr** [comment]
Hi @joprice, thanks for raising this issue, our current CLI uses `module-alias` library to resolve relative paths of files.
This is already an known issue tracked on their [Github](https://github.com/ilearnio/module-alias/issues/66).

**Comment by ThorneGladwin** [comment]
A temporary solution for this that will allow you to run ask-cli commands with `./node_modules/.bin/ask` is to install the `module-alias` library as a dev dependancy and then add the following to your package.json file:
`"_moduleAliases": {
    "@src": "./node_modules/ask-cli/lib",
    "@root": "./node_modules/ask-cli",
    "@test": "./node_modules/ask-cli/test"
  }`

This will then help to resolve references that the ask-cli uses such as `@src/utils/constants`. It doesn't solve this issue but at least allows you to work around it until it is fixed.

**Comment by btburton42** [comment]
@ThorneGladwin's suggestion is a solid workaround.

**Comment by EricWiener** [comment]
I found I had this issue when I first installed with `yarn` as `yarn global add ask-cli`. I wasn't able to get it working until I did:
```
yarn global remove ask-cli
sudo npm install -g ask-cli
```

**Comment by rdok** [comment]
> 
> 
> I found I had this issue when I first installed with `yarn` as `yarn global add ask-cli`. I wasn't able to get it working until I did:
> 
> ```
> yarn global remove ask-cli
> sudo npm install -g ask-cli
> ```

For best practice, [manually change npm’s default directory](https://docs.npmjs.com/resolving-eacces-permissions-errors-when-installing-packages-globally)

**Comment by kakhaUrigashvili** [comment]
fixed in v2.19.3

**Comment by gardnerjh** [comment]
@kakhaUrigashvili FWIW I still had this issue in v2.22.4, but was able to get around it using the workaround from @ThorneGladwin.
